src/Core/Domain/Common/Contracts/BaseEntityWith.cs

```diff
@@ -0,0 +1,22 @@
+﻿using MassTransit;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DN.WebApi.Domain.Common.Contracts;
+
+public abstract class BaseEntityWith<T> : BaseEntity

```

Comment:
Still don't see the need for another name...

What's wrong with something like this?

```c#
public abstract class BaseEntity : BaseEntity<Guid>
{
    protected BaseEntity() => Id = NewId.Next().ToGuid();
}

public abstract class BaseEntity<T>
{
    public T Id { get; protected set; } = default!;

    public List<DomainEvent> DomainEvents = new();
}
```

Reply:
Add some interfaces so they can be used e.g. in IRepositoryAsync as type constraints (IEntity in stead of BaseEntity):

```c#
public interface IEntity
{
}

public interface IEntity<T> : IEntity
{
    T Id { get; }
}

public abstract class BaseEntity<T> : IEntity<T>
{
    public T Id { get; protected set; } = default!;

    public List<DomainEvent> DomainEvents = new();
}

public abstract class BaseEntity : BaseEntity<Guid>
{
    protected BaseEntity() => Id = NewId.Next().ToGuid();
}

```

Reply:
I did something like that but due to inheritance relationship between `BaseEntity` & `BaseEnity<>` got some errors.
If you implement/improve this I would welcome your contributions,

Reply:
I'm working on other things at the moment... just trying to assist ;-)